dali/test/python/test_pipeline_debug.py

```diff
@@ -315,3 +315,38 @@ def test_init_config_pipeline():
     pipe_standard = init_config_pipeline(batch_size=8, num_threads=3, device_id=0)
     pipe_debug = init_config_pipeline(batch_size=8, num_threads=3, device_id=0, debug=True)
     compare_pipelines(pipe_standard, pipe_debug, 8, 10)
+
+
+@pipeline_def(batch_size=8, num_threads=3, device_id=0, seed=47, debug=True)
+def seed_pipeline():
+    coin_flip = fn.random.coin_flip()
+    normal = fn.random.normal()
+    uniform = fn.random.uniform()
+    batch_permutation = fn.batch_permutation()
+    return coin_flip, normal, uniform, batch_permutation
+
+
+def test_seed_generation():
+    pipe1 = seed_pipeline()
+    pipe2 = seed_pipeline()
+    compare_pipelines(pipe1, pipe2, 8, 10)
+
+
+@pipeline_def(batch_size=8, num_threads=3, device_id=0, seed=47, debug=True)
+def seed_rn50_pipeline_base():
+    rng = fn.random.coin_flip(probability=0.5)
+    jpegs, labels = fn.readers.file(
+        file_root=file_root, shard_id=0, num_shards=2)
```

Comment:
Maybe add random shuffle to the input data, that would be a good test for seed propagation.
```suggestion
        file_root=file_root, shard_id=0, num_shards=2, random_shuffle=True)
```

Reply:
done